cli/cli/.goreleaser.yml

```diff
@@ -5,24 +5,34 @@ before:
 
 builds:
   - id: cli
-    env:
-      # We need to set this to get a statically-linked binary, else we'll get a dynamically-linked binary that won't be able to be run anywhere
-      - CGO_ENABLED=0
-    main: .
     binary: "{{ .Env.CLI_BINARY_FILENAME }}"
+    main: .
     goos:
-      - linux
       - darwin
+      - linux
     goarch:
       - amd64
       - arm64
+    env:
+      - CGO_ENABLED=1
+      - CC={{ if index .Env "CC_WITH_DEFAULT" }}{{ .Env.CC_WITH_DEFAULT }}{{ else }} clang {{ end }}
```

Comment:
Add comment to explain what it `CC_WITH_DEFAULT`. Is this a Go release environment variable? Or does it come with the container?